main.tf

```diff
@@ -0,0 +1,692 @@
+provider "azurerm" {
+  features {}
+}
+
+# Create a resource group
+resource "azurerm_resource_group" "resource_group" {
+  name     = "resource-group-dmz"
+  location = "East US"
+}
+#create a Virtual Network.
+module "vnet" {
+  source = "./modules/vnet"
+  resource_group_name = "${azurerm_resource_group.resource_group.name}"
+  address_space       = ["10.0.0.0/16"]
+  location            = "${azurerm_resource_group.resource_group.location}"
+  name                = "dmzVnet"
+}
+
+# Create subnets.
+resource "azurerm_subnet" "webTierSubnet" {

```

Comment:
You may assume the label is used in conjunction with the resource name. So in webTierSubnet, Subnet is redundant because it will always be used with azurerm_subnet as a prefix. webTier is all thats needed.